### PR TITLE
Implement skeleton for getTransactionHistory

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement a skeleton of `getTransactionHistory` and tests inside of XpringClient.

Analogs:
JS: https://github.com/xpring-eng/Xpring-JS/pull/212
Java: https://github.com/xpring-eng/xpring4j/pull/116
Swift: https://github.com/xpring-eng/XpringKit/pull/124

### Context of Change

rippled recently implemented this RPC which we'd like to use.

Note that this functionality is not exposed in `XpringClient`, so this is still internal only. Currently `Transaction` objects are just empty containers. 

A future PR will:
- Update `Transaction` to have fields and to set them correctly from the protocol buffers
- Expose this as a public API

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

N/A - still internal only functionality. 

## Test Plan

New unit tests provided for CI.

### CODE REVIEW

File: src/main/java/io/xpring/xrpl/DefaultXpringClient.java [hunk 1]
```diff
@@ -204,6 +207,41 @@ public RawTransactionStatus getRawTransactionStatus(String transactionHash) thro
...
+        GetAccountTransactionHistoryResponse response = this.stub.getAccountTransactionHistory(request);
+
+        List<GetTransactionResponse> getTransactionResponseList = response.getTransactionsList();
+
+        List<ImmutableTransaction> transactionsList = getTransactionResponseList.stream()
+                .map(DefaultXpringClient::getTransactionResponseToImmutableTransaction)
+                .collect(Collectors.toList());
+
+        ImmutableTransaction [] transactions = new ImmutableTransaction[transactionsList.size()];
+        transactions = transactionsList.toArray(transactions);
```
Comment (nkramer44) [inline src/main/java/io/xpring/xrpl/DefaultXpringClient.java]: Could be wrong here, but I think you can skip the `List<ImmutableTransaction>` and go straight from a `Stream` to a `ImmutableTransaction[]`.

Looks something like:
```java
ImmutableTransaction[] transactions = getTransactionResponseList.stream()
                .map(DefaultXpringClient::getTransactionResponseToImmutableTransaction)
                .toArray(size -> new ImmutableTransaction[size])
```

**Comment by keefertaylor** [comment]
(\- other Java experts, \+ a devx-reviewer)

**Comment by hbergren** [comment]
Can you resolve the conflicts before I review?

**Comment by hbergren** [comment]
@keefertaylor , untagging myself as reviewer so Github Slack stops pestering me. Feel free to re-tag me when conflicts are fixed.

**Pull request closed without merging**